### ISSUE TITLE
Simple Variable Height?

### ISSUE THREAD

**Issue by ardalis** [thinkloop/knockout-js-infinite-scroll#13]
Can this approach be modified to support a variable height in the results, assuming we don't need to be able to jump to unrendered positions in the data set? I'm using the updateViewportDimensions function with itemHeight set to 71 currently. My data consists of 1-3 rows of text per table row, with 71px being the height of the longest value. I'm currently forcing all rows to use 71px for height as a workaround, which works but means a lot of wasted space (90% of the rows are 1 row of text, which would only require 31px; two rows requires 51px).

Can I set itemHeight to a function of the current row that would return the appropriate value (31,51,or 71)? Is there a relatively simple way to do this?

My requirements don't need to scale to 10s of 1000s of rows. The data involved is paging 30 rows at a time with max number of rows < 5000.

**Comment by thinkloop** [comment]
Hey,

What you're asking is definitely possible, but its not that straightforward the way this component is structured. For example, the function for determining the number of rows that fit on a page is:

``` javascript
props.numRowsPerPage = ko.computed(function() {
    var viewportHeight = parseInt(props.viewportHeight()),
        itemHeight = parseInt(props.itemHeight()) || -1;
    return Math.max(Math.ceil(viewportHeight / itemHeight), 0);
});
```

We simply divide viewportHeight by itemHeight and get how many items should fit on screen. With variable height, these calculations will have to loop through the specific items in question and add up their respective heights.

A more generalized way to do it would be to render the items first, then measure their actual height and record that in a separate collection of heights. Then use that to make the scroll calculations. There are challenges if the user resizes browser and such, but mostly surmountable.

Long story short - its doable, but takes some work, I am open to a PR if someone comes up with a good solution.
